src/Components/Shared/src/WebEventData.cs

```diff
@@ -240,23 +241,22 @@ private static ChangeEventArgs DeserializeChangeEventArgs(string eventArgsJson,
         {
             var changeArgs = Deserialize(eventArgsJson, jsonSerializerContext.ChangeEventArgs);
             var jsonElement = (JsonElement)changeArgs.Value!;
-            switch (jsonElement.ValueKind)
-            {
-                case JsonValueKind.Null:
-                    changeArgs.Value = null;
-                    break;
-                case JsonValueKind.String:
-                    changeArgs.Value = jsonElement.GetString();
-                    break;
-                case JsonValueKind.True:
-                case JsonValueKind.False:
-                    changeArgs.Value = jsonElement.GetBoolean();
-                    break;
-                default:
-                    throw new ArgumentException($"Unsupported {nameof(ChangeEventArgs)} value {jsonElement}.");
-            }
+            changeArgs.Value = GetChangeEventArgsValue(jsonElement);
+
             return changeArgs;
         }
+
+        private static object? GetChangeEventArgsValue(JsonElement jsonElement)
+            => jsonElement.ValueKind switch
+            {
+                JsonValueKind.Null => null,
+                JsonValueKind.String => jsonElement.GetString(),
+                JsonValueKind.True or JsonValueKind.False => jsonElement.GetBoolean(),
+                JsonValueKind.Array => jsonElement.EnumerateArray()
+                    .Select(GetChangeEventArgsValue)
```

Comment:
Is there a chance this is over-generalized? As far as I know, the only possible value types we'll receive here are strings, because the only element that can supply them is `<select>`, and its values are always strings. Is that correct or am I missing something?

It seems like if we did receive any nonstring values here, we should prefer to throw, since it's an indication that the client is doing something unsupported and maybe malicious (our client would never send nonstrings in an array). That being the case, we could produce a value of type `string[]` here rather than `object?[]`, and the logic in `MakeArrayTypeConverter` would be very slightly simplified.

Apologies if I'm misunderstanding the flow here!